lib/formatters/sarif.js

```diff
@@ -86,8 +87,24 @@ class SarifPrinter {
           }
         }
 
-        if (result.messages.length > 0) {
-          for (const message of result.messages) {
+        let fileErrors = result.messages;
+        let filteredMessages;
+
+        if (fileErrors.length > 0) {
+          let errorsFiltered = fileErrors.filter(
+            (error) => error.severity === Linter.ERROR_SEVERITY
+          );
+          let warnings = this.options.quiet
+            ? []
+            : fileErrors.filter((error) => error.severity === Linter.WARNING_SEVERITY);
+          let todos =
+            this.options.quiet || !this.options.includeTodo
+              ? []
+              : fileErrors.filter((error) => error.severity === Linter.TODO_SEVERITY);
+
+          filteredMessages = [...errorsFiltered, ...warnings, ...todos];
```

Comment:
Would this cause the results to be grouped by severity rather than file, or are they already grouped by severity before entering this code block? Just wondering if this will result in a meaningful change in the ordering.

As an alternative, you could do something like this:

```
let filteredMessages = fileErrors.filter((error) => {
  let isError = error.severity === Linter.ERROR_SEVERITY;
  let isWarning = error.severity === Linter.WARNING_SEVERITY;
  let isTodo = error.severity === Linter.TODO_SEVERITY;

  return isError || (!this.options.quiet && isWarning) || (!this.options.quiet && this.options.includeTodo && isTodo);
)};
```

Reply:
This is precisely the logic we use in the [json formatter](https://github.com/ember-template-lint/ember-template-lint/blob/a9bb2700e57d7d1f232a6c20e30cdaa50ae839c9/lib/formatters/json.js#L14-L28). Order isn't stated or guaranteed, and isn't materially relevant.